server/src/main/java/org/elasticsearch/indices/TermsLookup.java

```diff
@@ -82,7 +93,16 @@ public TermsLookup(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        out.writeString(type);
+        if (out.getVersion().onOrAfter(Version.V_7_0_0)) {
+            out.writeOptionalString(type);
+        } else {
+            if (type == null) {
+                throw new IllegalArgumentException("Typeless [terms] lookup queries are not supported if any " +
+                    "node is running a version before 7.0.");
```

Comment:
I don't really like throwing an IAE on the serialization layer, but I see where we don't have the information about the output streams version elsewhere. I will try to look into how this error propagates to e.g. the user making a rest call and I'd like to discuss the alternative of using a "dummy" type (with the downside that this then won't match lookup items with a non-default 6.x type name) again. In any case we should add something to the docs or some "known issues" document to capture this edge case.

Reply:
Great, I can also take a look at this by it out in REST tests. I will ping you offline to further discuss the idea of using a 'dummy' type name.